lib/std/Thread/Semaphore.zig

```diff
@@ -88,3 +88,16 @@ test "Thread.Semaphore" {
     sem.wait();
     try testing.expect(n == num_threads);
 }
+
+test "Thread.Semaphore - timedWait" {
+    var sem = Semaphore{};
+    try testing.expectEqual(@as(usize, @intCast(0)), sem.permits);
```

Comment:
You don't need the casting, just:

    try std.testing.expectEqual(0, sem.permits);

is sufficient (and below too).